python/dataset_plotting.py

```diff
@@ -269,10 +257,24 @@ def plot_image(input_data, axes=None, contours=False, plot=True, logcb=False, cb
             yaxis = dict(title = axis_label(ycoord))
             )
 
+        # Check if dimensions of arrays agree, if not, plot the transpose
+        zlabs = zdims.labels
+        if (zlabs[0] == xdims.labels[0]) and (zlabs[1] == ydims.labels[0]):
```

Comment:
What if the data is square but transposed?

Reply:
Not sure such a situation could arise? If the data in the dataset has been transposed then it should be reflected in the axis labels. If the array has been transposed and the labels not changed then we have no way of knowing that from the plotting routine